client/components/signup/components/login/login.js

```diff
@@ -9,30 +12,55 @@ class Login extends PureComponent {
 		gotoRegister: PropTypes.func.isRequired
 	}
 
-	handleEl = null
-
 	state = {
 		handle: '',
-		password: ''
+		password: '',
+		showErrMsg: false,
+		errMsg: ''
 	}
 
+	handleEl = null
+	errorIndex = 0
+
 	componentDidMount () {
 		this.handleEl.focus()
 	}
 
+	hideErrMsg = () => {
+		this.setState({ showErrMsg: false })
+	}
+
 	handleGenericInput = evt => {
 		const tgt = evt.target
 		const value = tgt[tgt.type === 'checkbox' ? 'checked' : 'value']
 
 		this.setState({ [tgt.name]: value })
 	}
 
-	handleSubmit = evt => {
+	handleSubmit = async evt => {
 		evt.preventDefault()
 
 		const { handle, password } = this.state
 
-		this.props.onSubmit(handle, password)
+		const res = await this.props.onSubmit(handle, password)
+
+		if (!res.success) {
+			this.setState({
+				showErrMsg: true,
+				errMsg: res.message || 'An unknown error occurred.'
+			}, () => {
+				this.errorIndex++
+
+				const thisIndex = this.errorIndex
+
+				sleep(3000)
+					.then(() => {
+						if (thisIndex !== this.errorIndex) return
```

Comment:
You should add braces after this `if` statement for increased clarity.

Reply:
I disagree. I think if there's only a single statement then doing if/else if/else without braces is more terse and ultimately more readable, particularly when doing a quick check and return, a code habit I use a lot. Unlike in other situations, such as where returning from an arrow function (due to lack of braces) at least affects how you contextually read the code, I don't think the braces would serve any purpose here.

It just comes down to code style I suppose.